### PR TITLE
Component Test For Components Model - Transceivers

### PR DESCRIPTION
**Pull request by niranjanivivek**
This PR is created to add component test case for OpenConfig platform components root model transceivers

Note:
Dependency on below sonic-mgmt-common PRs.

Base PR - Support for Components Model Root - https://github.com/sonic-net/sonic-mgmt-common/pull/224
Incremental PR -  Support for Components Model - Transceiver - https://github.com/sonic-net/sonic-mgmt-common/pull/226

**Test Output**
=== RUN   TestGnmiGetTranslibPfmCompXcvr
=== RUN   TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr
    server_test.go:3367: [TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr] gotVal:
        {
          "openconfig-platform:component": [
            {
              "config": {
                "name": "Ethernet0"
              },
              "name": "Ethernet0",
              "openconfig-platform-transceiver:transceiver": {
                "physical-channels": {
                  "channel": [
                    {
                      "config": {
                        "index": 0
                      },
                      "index": 0,
                      "state": {
                        "index": 0,
                        "input-power": {
                          "instant": "-Inf"
                        },
                        "laser-bias-current": {
                          "instant": "0"
                        },
                        "tx-laser": false
                      }
                    },
                    {
                      "config": {
                        "index": 1
                      },
                      "index": 1,
                      "state": {
                        "index": 1,
                        "input-power": {
                          "instant": "-Inf"
                        },
                        "laser-bias-current": {
                          "instant": "0"
                        },
                        "tx-laser": false
                      }
                    },
                    {
                      "config": {
                        "index": 2
                      },
                      "index": 2,
                      "state": {
                        "index": 2,
                        "input-power": {
                          "instant": "-Inf"
                        },
                        "laser-bias-current": {
                          "instant": "0"
                        },
                        "tx-laser": false
                      }
                    },
                    {
                      "config": {
                        "index": 3
                      },
                      "index": 3,
                      "state": {
                        "index": 3,
                        "input-power": {
                          "instant": "-Inf"
                        },
                        "laser-bias-current": {
                          "instant": "0"
                        },
                        "tx-laser": false
                      }
                    }
                  ]
                }
              },
              "state": {
                "empty": false,
                "firmware-version": "D",
                "hardware-version": "D",
                "mfg-date": "2011-12-13 ",
                "mfg-name": "Amphenol",
                "name": "Ethernet0",
                "oper-status": "openconfig-platform-types:ACTIVE",
                "part-no": "599690001",
                "removable": true,
                "serial-no": "APF11490011J7E",
                "temperature": {
                  "instant": "101"
                },
                "type": "openconfig-platform-types:TRANSCEIVER"
              }
            }
          ]
        }
=== RUN   TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_State
    server_test.go:3367: [TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_State] gotVal:
        {
          "openconfig-platform:state": {
            "empty": false,
            "firmware-version": "D",
            "hardware-version": "D",
            "mfg-date": "2011-12-13 ",
            "mfg-name": "Amphenol",
            "name": "Ethernet0",
            "oper-status": "openconfig-platform-types:ACTIVE",
            "part-no": "599690001",
            "removable": true,
            "serial-no": "APF11490011J7E",
            "temperature": {
              "instant": "101"
            },
            "type": "openconfig-platform-types:TRANSCEIVER"
          }
        }
=== RUN   TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_Temperature
    server_test.go:3367: [TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_Temperature] gotVal:
        {
          "openconfig-platform:instant": "101"
        }
=== RUN   TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_Presence
    server_test.go:3367: [TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_Presence] gotVal:
        {
          "openconfig-platform:empty": false
        }
--- PASS: TestGnmiGetTranslibPfmCompXcvr (1.25s)
    --- PASS: TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr (0.04s)
    --- PASS: TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_State (0.02s)
    --- PASS: TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_Temperature (0.02s)
    --- PASS: TestGnmiGetTranslibPfmCompXcvr/Get_OC_Platform_Comp_Xcvr_Presence (0.02s)
PASS

### CODE REVIEW

**Comment by niranjanivivek** [comment]
@ndas7  and @rlucus  Please review and approve

File: gnmi_server/server_test.go [hunk 1]
```diff
@@ -7583,3 +7583,164 @@ func TestServeUDSErrorDoesNotStopTCP(t *testing.T) {
...
+
+	listFields := map[string]interface{}{
+		"name": "Ethernet0",
+	}
+	listStatusFields := map[string]interface{}{
+		"name":   "Ethernet0",
+		"status": true,
+	}
+
+	// Explicitly write both delimiter schemas into DB 0
```
Comment (ndas7) [inline gnmi_server/server_test.go]: These state tables are not present in Config DB, please remove

File: gnmi_server/server_test.go [hunk 2]
```diff
@@ -7583,3 +7583,164 @@ func TestServeUDSErrorDoesNotStopTCP(t *testing.T) {
...
+		operation     op_t
+		valTest       bool
+	}{
+		{
+			desc:          "Plfm xvr delete",
+			pathTarget:    "OC_YANG",
+			textPbPath:    pathToPb("openconfig-platform:components/component[name=Ethernet0]/"),
+			attributeData: "",
+			wantRetCode:   codes.Unknown,
+			operation:     Delete,
```
Comment (ndas7) [inline gnmi_server/server_test.go]: We do not need to support Set delete for transceivers, please remove

File: gnmi_server/server_test.go [hunk 3]
```diff
@@ -7583,3 +7583,164 @@ func TestServeUDSErrorDoesNotStopTCP(t *testing.T) {
...
+		valTest     bool
+	}{
+		{
+			desc:       "Get OC Platform Comp Xcvr",
+			pathTarget: "OC_YANG",
+			textPbPath: `
+                        elem: <name: "openconfig-platform:components" > elem: <name: "component" key:<key:"name" value:"Ethernet0" > >
+                `,
+			wantRetCode: codes.OK,
+			wantRespVal: emptyRespVal,
```
Comment (ndas7) [inline gnmi_server/server_test.go]: There should be more tests to test for the actual values in the Get call for transceivers, not just empty values.

File: gnmi_server/server_test.go [hunk 4]
```diff
@@ -7583,3 +7583,164 @@ func TestServeUDSErrorDoesNotStopTCP(t *testing.T) {
...
+	configClient.HSet(ctx, "TRANSCEIVER_CFG|Ethernet0", listFields)
+	configClient.HSet(ctx, "TRANSCEIVER_INFO|Ethernet0", listFields)
+	configClient.HSet(ctx, "TRANSCEIVER_STATUS|Ethernet0", listStatusFields)
+	configClient.HSet(ctx, "TRANSCEIVER_DOM_SENSOR|Ethernet0", listFields)
+
+	stateClient := getRedisClientN(t, 6, ns) // Force target DB 0
+	defer stateClient.Close()
+
+	// Explicitly write both delimiter schemas into DB 0
+	stateClient.HSet(ctx, "TRANSCEIVER_INFO|Ethernet0", listFields)
```
Comment (ndas7) [inline gnmi_server/server_test.go]: Please make sure to address Ryan's comments from #738 in this PR as well - including resetting these tables in the DB

**Review by ndas7 — CHANGES_REQUESTED** [review]

**Review by ndas7 — APPROVED** [review]
